### ISSUE TITLE
Declared dependencies do not seem to be provided to the generator

### ISSUE THREAD

**Issue by ryan-gustafson** [opengl-8080/assertjGen-gradle-plugin#7]
I get this error (included at the end) when trying to generate assertions for a class which happens to use ANTLR dependencies.  The antlr-runtime is setup properly in my Gradle configurations, and available in the classpath for other tasks (e.g. compiling and testing).

```
// A portion of the dependencies...
dependencies {
    implementation "org.antlr:antlr4-runtime:4.7"
}
```

Does this plugin provide those dependencies to the generator?

I ran Gradle with --debug and I see the call to launch the `org.assertj.assertions.generator.cli.AssertionGeneratorLauncher`, but it would appear to only include the 
`build\classes\main` directory and none of my external dependencies?

Lastly, the error message:

```
20:18:37.237 [ERROR] [system.err] Exception in thread "main" java.lang.NoClassDefFoundError: org/antlr/v4/runtime/TokenSource
20:18:37.238 [ERROR] [system.err] 	at java.lang.Class.forName0(Native Method)
20:18:37.238 [ERROR] [system.err] 	at java.lang.Class.forName(Class.java:348)
20:18:37.238 [ERROR] [system.err] 	at org.assertj.assertions.generator.util.ClassUtil.loadClass(ClassUtil.java:234)
20:18:37.238 [ERROR] [system.err] 	at org.assertj.assertions.generator.util.ClassUtil.tryToLoadClass(ClassUtil.java:227)
20:18:37.239 [ERROR] [system.err] 	at org.assertj.assertions.generator.util.ClassUtil.collectClasses(ClassUtil.java:95)
20:18:37.239 [ERROR] [system.err] 	at org.assertj.assertions.generator.util.ClassUtil.collectClasses(ClassUtil.java:77)
20:18:37.239 [ERROR] [system.err] 	at org.assertj.assertions.generator.cli.AssertionGeneratorLauncher.main(AssertionGeneratorLauncher.java:54)
20:18:37.239 [ERROR] [system.err] Caused by: java.lang.ClassNotFoundException: org.antlr.v4.runtime.TokenSource
20:18:37.239 [ERROR] [system.err] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
20:18:37.239 [ERROR] [system.err] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
20:18:37.239 [ERROR] [system.err] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
20:18:37.239 [ERROR] [system.err] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
20:18:37.240 [ERROR] [system.err] 	... 7 more
```

**Comment by opengl-8080** [comment]
Thank you for your report.

However, I can't have this error in my environment.  
(I made a simple project -> [sample_project.zip](https://github.com/opengl-8080/assertjGen-gradle-plugin/files/952075/simple_project.zip))

Can you create a simple project to confirm this error?

**Comment by ryan-gustafson** [comment]
Try adding the following to the `Hoge.java` class, all I saw was an unused import in your sample:

```
public static void foo(TokenSource tokenSource) {
    System.out.println(tokenSource);
}
```

I get:
```
Working Directory: C:\software\workspace\assertjGen
Gradle User Home: C:\Users\Ryan\.gradle
Gradle Distribution: Gradle wrapper from target build
Gradle Version: 3.4.1
Java Home: C:\Java\jdk1.8.0_121
JVM Arguments: None
Program Arguments: None
Gradle Tasks: build

:compileJava
:processResources NO-SOURCE
:classes
:jar
:assemble
:assertjClean
:assertjGen
23:45:32.324 INFO  o.a.a.g.c.AssertionGeneratorLauncher - Generating assertions for classes [class sample.Hoge]
23:45:32.375 INFO  o.a.a.g.c.AssertionGeneratorLauncher - Generating assertions for class : sample.Hoge
Exception in thread "main" java.lang.NoClassDefFoundError: org/antlr/v4/runtime/TokenSource
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetPublicMethods(Class.java:2902)
	at java.lang.Class.getMethods(Class.java:1615)
	at org.assertj.assertions.generator.util.ClassUtil.getterMethodsOf(ClassUtil.java:349)
	at org.assertj.assertions.generator.description.converter.ClassToClassDescriptionConverter.getterDescriptionsOf(ClassToClassDescriptionConverter.java:57)
	at org.assertj.assertions.generator.description.converter.ClassToClassDescriptionConverter.convertToClassDescription(ClassToClassDescriptionConverter.java:47)
	at org.assertj.assertions.generator.cli.AssertionGeneratorLauncher.toClassDescription(AssertionGeneratorLauncher.java:102)
	at org.assertj.assertions.generator.cli.AssertionGeneratorLauncher.generateFlatAssertions(AssertionGeneratorLauncher.java:95)
	at org.assertj.assertions.generator.cli.AssertionGeneratorLauncher.main(AssertionGeneratorLauncher.java:59)
Caused by: java.lang.ClassNotFoundException: org.antlr.v4.runtime.TokenSource
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 10 more
:assertjGen FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assertjGen'.
> Process 'command 'C:\Java\jdk1.8.0_121\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2.132 secs
```

**Comment by opengl-8080** [comment]
Thank you.
I get this error.

I will try to consider how to modify this problem.

**Comment by opengl-8080** [comment]
I fixed [a bug #4](https://github.com/opengl-8080/assertjGen-gradle-plugin/issues/4).

Therefore, you can configure build.gradle like following to escape this problem.

```groovy
dependencies {
    implementation "org.antlr:antlr4-runtime:4.7"
    assertj configurations.implementation
}
```

Please update plugin version to 1.1.3.

**Comment by marekadare** [comment]
I have a similar problem.

:compileJava
:generateGitProperties
:processResources
:classes
:jar
:execJar
:findMainClass
:bootRepackage
:assertjClean
Exception in thread "main" java.lang.NoClassDefFoundError: javax/servlet/Filter
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at com.google.common.reflect.ClassPath$ClassInfo.load(ClassPath.java:266)
	at org.assertj.assertions.generator.util.ClassUtil.getPackageClassesFromClasspathJars(ClassUtil.java:134)
	at org.assertj.assertions.generator.util.ClassUtil.getClassesInPackage(ClassUtil.java:122)
	at org.assertj.assertions.generator.util.ClassUtil.collectClasses(ClassUtil.java:100)
	at org.assertj.assertions.generator.util.ClassUtil.collectClasses(ClassUtil.java:77)
	at org.assertj.assertions.generator.cli.AssertionGeneratorLauncher.main(AssertionGeneratorLauncher.java:54)
Caused by: java.lang.ClassNotFoundException: javax.servlet.Filter
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 18 more
:assertjGen FAILED

Thanks in advance
